src/crimson/net/SocketConnection.h

```diff
@@ -83,6 +83,10 @@ class SocketConnection : public Connection {
 
 #ifdef UNIT_TESTS_BUILT
   bool is_closed() const override;
+
+  bool is_winner_side() const override;
```

Comment:
IMHO, `is_winner_side()` is difficult to interpret. as a connection does not represent either side of it. probably we can change it to "peer_wins" or "local_wins" ? i'd prefer to "local_wins". and change the body of the method to:

```c++
messenger.get_myaddr() <= peer_addr || policy.server
```

Reply:
I'm not sure if we are in the same page, because IMHO even if it is renamed to `local_wins()`, the body of the method would be still:
`peer_addr > messenger.get_myaddr() && !policy.server`

Look at a typical racing case for example:
```
<peer A>               <peer B>
conn-A-connecting  ==>  conn-accepting-A
conn-accepting-B   <==  conn-B-connecting
```

From the current logic, either side (A or B) would know if it is the winner by looking at:
`addr(peer) < addr(me) && not_server(me)`

Which means if peer A is the winner, both `is_winner_side(conn-A-connecting)` an`is_winner_side(conn-accepting-B)` would be true, and thus A's connecting attempt of `conn-A-connecting` should not be replaced by `conn-accepting-B`.

As for peer B, both `is_winner_side(conn-B-connecting)` and `is_winner_side(conn-accepting-A)` are false, and thus B's connecting attempt of `conn-B-connecting` would be replaced by `conn-accepting-A`.

What do you think?

Reply:
sorry for the confusion. i meant to say "peer_wins". the reason i suggested
```c++
messenger.get_myaddr() <= peer_addr || policy.server
```

was to avoid negative logic in condition statement.

Reply:
Changed to `peer_wins()`, but I think the body should be:
```
messenger.get_myaddr() > peer_addr || policy.server
```

Normally `messenger.get_myaddr()` won't be equal to `peer_addr`, because this would be the case of loopback connection.